lib/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,213 @@
+  const styles = useStyles(styleRefs);
+
+  return (
+    <Box
+      component="span"
+      display="flex"
+      alignItems="center"
+      justifyContent="center"
+      height="touchable"
+      width="touchable"
+      position="relative"
+      className={styles.hover}
+    >
+      <Overlay
+        component="span"
+        background={current ? 'formAccent' : 'selection'}
+        transition={current ? undefined : 'fast'}
+        borderRadius="standard"
+        className={[styles.background, current ? styles.current : undefined]}
+      />
+      <Box component="span" zIndex={1} userSelect="none">
+        <Text
+          component="span"
+          baseline={false}
+          align="center"
+          weight={current ? 'medium' : undefined}
+          tone={current ? 'formAccent' : undefined}
+        >
+          {number}
+        </Text>
+      </Box>
+    </Box>
+  );
+};
+
+const shouldHideOnTablet = (number: number, page: number, total: number) => {
+  if (page === 1) {
+    return number > page + 2;
+  }
+
+  if (page === total) {
+    return number < page - 2;
+  }
+
+  return page < number - 1 || page > number + 1;
+};
+
+export const Pagination = ({
+  page,
+  total,
+  linkProps,
+  label,
+  pageLabel = (p: number) => `Go to page ${p}`,
+  nextLabel = 'Next',
+  previousLabel = 'Previous',
+}: PaginationProps) => {
+  const pages = paginate({ page, total });
+  const showPrevious = total > pages.length && page > 1;
+  const showNext = total > pages.length && page < total;
+
+  return (
+    <Box component="nav" aria-label={label}>
+      <Box component="ul">
+        <Columns space={['none', 'xsmall']} align="center">
+          <Column>
+            <Box
+              display="flex"
+              justifyContent="flexEnd"
+              component="li"
+              opacity={showPrevious ? undefined : 0}
+              pointerEvents={showPrevious ? undefined : 'none'}
+              transition="fast"
+            >
+              <Link
+                {...linkProps(page - 1)}
+                rel="prev"
+                aria-label={previousLabel}
+                aria-hidden={!showPrevious}
+                aria-disabled={!showPrevious}
+                tabIndex={showPrevious ? undefined : -1}
+                title={previousLabel}
+              >
+                <PageNav label={previousLabel} direction="prev" />
+              </Link>
+            </Box>
+          </Column>
+
+          <Column width="content">
+            <Box display="flex">
+              {pages.map((pageNumber) => {
+                const current = page === pageNumber;
+
+                return (
+                  <Hidden
+                    below={
+                      shouldHideOnTablet(pageNumber, page, total)
+                        ? 'tablet'
+                        : undefined
+                    }
+                    key={pageNumber}
+                  >
+                    <Box component="li">
```

Comment:
The HTML might need some work here since the `<li>` elements aren't direct children of the `<ul>`.